Hunted Mobile/Hunted Mobile/Hunted Mobile/Service/Json/PlayerJsonService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Hunted_Mobile.Model;
+using Hunted_Mobile.Model.GameModels;
+using Hunted_Mobile.Model.Response.Json;
+
+using System;
+
+namespace Hunted_Mobile.Service.Json {
+    public class PlayerJsonService : JsonConversionService<Player, UserData> {
+        public PlayerJsonService() {
+        }
+
+        public override string ToJson(Player player) {
+            return ConvertToJson(new UserData {
+                id = player.Id,
+                username = player.UserName,
+                location = player.Location.ToCsvString(),
+                status = player.Status,
+                caught_at = player is Thief ? ((Thief) player).CaughtAt : null,
+                role = player.InviteKey.Role
+            });
+        }
+
+        public override Player ToObject(UserData data) {
+            Player player = new Player() {
+                Id = data.id,
+                InviteKey = new InviteKey() {
+                    Role = data.role,
+                    UserId = data.id
+                },
+                Location = new LocationJsonService().ToObjectFromCsv(data.location),
+                Status = data.status,
+                UserName = data.username
+            };
+
+            if(data.role == "thief") {
```

Comment:
Zeker dat je hiervoor geen enum wilt gebruiken? De tekst wordt meerdere keren gebruikt en daarbij kunnen er makkelijk spellingsfouten worden gemaakt. Met een enum is het meteen duidelijk welke rollen er zijn en kan dat nooit misgaan.

Reply:
De enum zou een verbetering zijn idd, maar deze bestaat nog niet in mijn branch. Nog beter zou denk ik een PlayerFactory zijn, die een speler creëert gebaseerd op een `string role`.